Lib/ufo2ft/util.py

```diff
@@ -99,6 +127,45 @@ def newGlyph(name):
     return glyphSet
 
 
+def deepCopyContours(
+    glyphSet, parent, composite, transformation, specificComponents=None
+):
+    """Copy contours from component to parent, including nested components."""
+
+    for nestedComponent in composite.components:
+        if specificComponents and nestedComponent.baseGlyph not in specificComponents:
+            continue
+
+        try:
+            nestedBaseGlyph = glyphSet[nestedComponent.baseGlyph]
+        except KeyError:
+            logger.warning(
+                "dropping non-existent component '%s' in glyph '%s'",
+                nestedComponent.baseGlyph,
+                parent.name,
+            )
+        else:
+            deepCopyContours(
+                glyphSet,
+                parent,
+                nestedBaseGlyph,
+                transformation.transform(nestedComponent.transformation),
+            )
```

Comment:
when you recurse here, don't you also need to pass on that `specificComponents` argument?

Reply:
Oops, probably 😁

Reply:
Np. A test that catches that would be nice too